src/index.rs

```diff
@@ -47,6 +53,21 @@ impl Index {
 
         let field_id =
             unsafe { raw::RediSearch_CreateField(self.inner, name.as_ptr(), ftype.bits, fopt) };
+        unsafe {
+            raw::RediSearch_TextFieldSetWeight(self.inner, field_id, weight);
+            if let Some(options) = tag_options {
+                if let Some(sperator) = options.tag_separator {
+                    raw::RediSearch_TagFieldSetSeparator(self.inner, field_id, sperator as i8);
```

Comment:
You should use `c_char` here instead of `i8` (C `char` can be signed or unsigned depending on the platform).

Reply:
:+1: